nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/realm/certificate/CertificateRealm.java

```diff
@@ -162,11 +168,27 @@ public String getAuthType() {
      * @param principal The Principal object from the user certificate.
      *
      */
-    public void authenticate(Subject subject, Principal principal) {
+    public String authenticate(Subject subject, Principal principal) {
```

Comment:
At a glance, just wondering if this actually works? The common name is returned here, but not installed in the security context. The returned name is only used for auditing. Am I missing something?

Reply:
It does work against the written tests in ee7 and against the effective reproducer that I've been working with. It's a change to meet a request made by Ondrej for the sakes of consistency across the authentication logic

Reply:
@cubastanley I did overlook something indeed, and that's that `name` is used  as the first argument of the security context, my bad, sorry. So that makes it work wrt role mapping. What I was actually looking at however was setting the principal in the Subject to one with the CN as the name". This is pending clarification of the original issue.